code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -471,3 +471,85 @@
 		M.emote(pick("twitch","laugh","frown"))
 	..()
 	. = 1
+
+//I had to do too much research on this to make this a thing. Hopefully the FBI won't kick my door down.
+/datum/reagent/drug/ketamine
+	name = "Ketamine"
+	description = "A heavy duty tranquilizer found to also invoke feelings of euphoria, and assist with pain. Popular at parties and amongst small frogmen who drive Honda Civics."
+	reagent_state = LIQUID
+	color = "#c9c9c9"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	addiction_threshold = 8
+	overdose_threshold = 16
+
+/datum/reagent/drug/ketamine/on_mob_metabolize(mob/living/L)
+	ADD_TRAIT(L, TRAIT_IGNOREDAMAGESLOWDOWN, type)
+	. = ..()
+
+/datum/reagent/drug/ketamine/on_mob_delete(mob/living/L)
+	REMOVE_TRAIT(L, TRAIT_IGNOREDAMAGESLOWDOWN, type)
+	. = ..()
+
+/datum/reagent/drug/ketamine/on_mob_life(mob/living/carbon/M)
+	//Friendly Reminder: Ketamine is a tranquilizer and will sleep you.
+	switch(current_cycle)
+		if(10)
+			to_chat(M, "<span class='warning'>You start to feel tired...</span>" )
+		if(11 to 25)
+			M.drowsyness ++
+		if(26 to INFINITY)
+			M.Sleeping(60, 0)
+			. = 1
+	//Providing a Mood Boost
+	M.confused -= 3
+	M.jitteriness -= 5
+	M.disgust -= 3
+	//Ketamine is also a dissociative anasthetic which means Hallucinations!
+	M.hallucination = 5
```

Comment:
```suggestion
	M.hallucination = max(M.hallucination, 5)
```
`M.hallucination = 5` doesn't *add* hallucination, it will set hallucination to 5 no matter what. Hallucination of 5 isn't that strong, and it'll *override* other hallucinogens. You also can do this:

```suggestion
	M.hallucination += 5
```